### PR TITLE
Query updates

### PR DESCRIPTION
**Pull request by nuurcodes**

### CODE REVIEW

File: src/api/blockchain/crucibles.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+              value: { amount: lockData.balance },
+              address: lockData.token,
+              symbol: tokenSymbol,
+              imageUrl
+            },
+            subscriptionCount: vaultData.stakes.length
+          };
+        }
+      })
+    )) as CrucibleLock[];
```
Comment (itirabasso) [inline src/api/blockchain/crucibles.ts:157]: would it help if we have a contract deployed on mainnet to read the state of a bunch of crucibles at once?

Comment (itirabasso) [reply src/api/blockchain/crucibles.ts:157]: something like this:

```
    function getLocksByDelegate(
        address crucible,
        address delegate,
        uint256[] memory indices
    ) external view returns (IUniversalVault.LockData[] memory) {
        require(indices.length > 0, "at least one index");
        uint256 len = 0;

        
        IUniversalVault.LockData[] memory locks = new IUniversalVault.LockData[](indices.length);
        for (uint256 i = 0; i < indices.length - 1; i++) {
            locks[i] = IUniversalVault(crucible).getLockAt(indices[i]);
        }

        IUniversalVault.LockData[] memory ret = new IUniversalVault.LockData[](locks.length);
        for (uint256 i = 0; i < locks.length - 1; i++) {
            IUniversalVault.LockData memory lock = locks[i];
            if (lock.delegate == delegate) {
                ret[ret.length] = lock;
            }
        }
        return locks;
    }
```

Comment (itirabasso) [reply src/api/blockchain/crucibles.ts:157]: My question follows our discord conversation about the loading times. Can we improve the loading times if we have a contract that reads the crucibles state in bulk?

Comment (nuurcodes) [reply src/api/blockchain/crucibles.ts:157]: this would be really awesome!

**Pull request closed without merging**